sktime/forecasting/patchtst_forecaster.py

```diff
@@ -0,0 +1,507 @@
+                .to_absolute(self._cutoff)
+                ._values.tolist()
+                * pred.shape[0]
+            )
+            index = pd.MultiIndex.from_arrays(
+                ins + [idx],
+                names=y.index.names,
+            )
+        else:
+            index = (
+                ForecastingHorizon(range(1, pred.shape[1] + 1))
+                .to_absolute(self._cutoff)
+                ._values
+            )
+
+        df_pred = pd.DataFrame(
+            # batch_size * num_timestams, n_cols
+            pred.reshape(-1, pred.shape[-1]),
+            index=index,
+            columns=y_columns,
+        )
+
+        absolute_horizons = fh.to_absolute_index(self.cutoff)
+        dateindex = df_pred.index.get_level_values(-1).map(
+            lambda x: x in absolute_horizons
+        )
+        df_pred = df_pred.loc[dateindex]
+        df_pred.index.names = y_index_names
+        return df_pred
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        params_set = []
+        params1 = {
+            "random_mask_ratio": 0.4,
+            "d_model": 64,
+            "num_attention_heads": 4,
+            "ffn_dim": 32,
+            "head_dropout": 0.2,
+            "batch_size": 16,
+            "epochs": 1,
+            "validation_split": 0.0,
+        }
+        params_set.append(params1)
+        params2 = {
+            "d_model": 128,
+            "num_attention_heads": 4,
+            "ffn_dim": 32,
+            "head_dropout": 0.2,
+            "batch_size": 16,
+            "epochs": 1,
+            "validation_split": 0.0,
+        }
+        params_set.append(params2)
+
+        return params_set
+
+
+def _same_index(data):
+    data = data.groupby(level=list(range(len(data.index.levels) - 1))).apply(
+        lambda x: x.index.get_level_values(-1)
+    )
+    assert data.map(
+        lambda x: x.equals(data.iloc[0])
+    ).all(), "All series must has the same index"
+    return data.iloc[0], len(data.iloc[0])
+
+
+def _frame2numpy(data):
+    idx, length = _same_index(data)
+    arr = np.array(data.values, dtype=np.float32).reshape(
+        (-1, length, len(data.columns))
+    )
+    return arr
+
+
+class PyTorchDataset(Dataset):
```

Comment:
can you make this class private? Alternatively, consider using sth from the other `torch` interfaces.

Reply:
the other `PytorchDataset` classes are public in other global forecasters, is there any particular reason why it should be private in this one? Just curious

Reply:
> Alternatively, consider using sth from the other `torch` interfaces.

I'm not quite sure what you mean by this, can you please elaborate?

Reply:
> Alternatively, consider using sth from the other `torch` interfaces.

I don't think using `PytorchDataset` class from some other model is a good idea here, because right now each dataset class is created for their respective models. Although several `PytorchDataset` classes are much identical, yet they should be refactored and coupled in another PR.

Reply:
Yes, for example, the `PytorchDataset` class in `momentfm` cannot be the same as the one used in here or in lets say `ttm`.